modules/drivers/sqlite/test/metabase/test/data/sqlite.clj

```diff
@@ -44,6 +44,13 @@
     (when (#{:count :cum-count} ag-type)
       {:base_type :type/Integer}))))
 
+(defmethod sql.tx/create-table-sql :sqlite
+  [driver dbdef table-def]
+  ;; Sqlite doesn't automatically create an index for pk, so we need to manually create one
```

Comment:
Can you explain _why_ we need to manually create one?

Reply:
the tests in `metabase.driver.sql-jdbc.sync.describe-table-test/describe-table-indexes-advanced-index-type-test` assumes all id columns are indexed.

this is to comply with that.

Reply:
If I'm reading the documentation correctly SQLite ([here](https://www.sqlite.org/lang_createtable.html#:~:text=3.6.%20UNIQUE%20constraints) and [here](https://www.sqlite.org/lang_createindex.html#:~:text=DROP%20INDEX%20command.-,1.1.%20Unique%20Indexes,-If%20the%20UNIQUE)), a single primary key column is indexed. Same with any column with a `UNIQUE` constraint.

If `.getIndexInfo` does not return them, then `describe-table-indexes` should have a special implementation that returns them anyway.

Reply:
can you refer to the part where it says the PK is indexed?

either way I did a check and it doesn't create an index by default.

```
sqlite> CREATE TABLE user ( id INTEGER PRIMARY KEY, name TEXT);


sqlite> CREATE INDEX idx_user_name ON user (name);CREATE INDEX idx_user_name ON user (name);
sqlite> PRAGMA index_list(user);
0|idx_user_name|0|c|0


sqlite> CREATE INDEX idx_user_id ON user (id);
sqlite> PRAGMA index_list(user);
0|idx_user_id|0|c|0
1|idx_user_name|0|c|0
sqlite>
```

Reply:
> can you refer to the part where it says the PK is indexed?

[here](https://www.sqlite.org/lang_createtable.html#:~:text=3.6.%20UNIQUE%20constraints)
```
In most cases, UNIQUE and PRIMARY KEY constraints are implemented by creating a unique index in the database. (The exceptions are [INTEGER PRIMARY KEY](https://www.sqlite.org/lang_createtable.html#rowid) and PRIMARY KEYs on [WITHOUT ROWID](https://www.sqlite.org/withoutrowid.html) tables.) Hence, the following schemas are logically equivalent:

CREATE TABLE t1(a, b UNIQUE);

CREATE TABLE t1(a, b PRIMARY KEY);

CREATE TABLE t1(a, b);
CREATE UNIQUE INDEX t1b ON t1(b);
```

and in the documentation for [unique index](https://www.sqlite.org/lang_createindex.html#:~:text=DROP%20INDEX%20command.-,1.1.%20Unique%20Indexes,-If%20the%20UNIQUE) there's nothing saying a "unique index isn't a real index" or anything like that.

> either way I did a check and it doesn't create an index by default.

That's because it's an `INTEGER PRIMARY KEY`
https://stackoverflow.com/questions/55036387/why-would-pragma-schema-index-listtable-name-not-show-primary-keys

Reply:
`index_list` does return an index for the primary key if it's `TEXT PRIMARY KEY`.

```
sqlite> CREATE TABLE user ( id TEXT PRIMARY KEY, name TEXT);
sqlite> PRAGMA index_list(user);
0|sqlite_autoindex_user_1|1|pk|0
```

PRIMARY KEY and unique constraints are included in the documentation for `index_list` [here](https://www.sqlite.org/pragma.html#:~:text=PRAGMA%20schema.index_list(table%2Dname)%3B).

Reply:
I think the correct behaviour here is to make a special case for sqlite integer primary keys.

And we should also mark columns that are `UNIQUE` as indexed too. I think you might just need to add a test for that though, because your implementation might already handle it.